### PR TITLE
Support sentinel notification-script and client-reconfig-script

### PR DESCRIPTION
**Pull request by matkam**

### CODE REVIEW

File: templates/default/sentinel.conf.erb [hunk 1]
```diff
@@ -117,6 +117,7 @@ sentinel failover-timeout <%=@name%> <%=@failovertimeout%>
 # Example:
 #
 # sentinel notification-script mymaster /var/redis/notify.sh
+<%= "sentinel notification-script #{@name} #{@notification_script}" unless @notification_script.nil? %>
```
Comment (martinb3) [inline templates/default/sentinel.conf.erb]: Should this just be guarded on `unless @notification_script`?

File: templates/default/sentinel.conf.erb [hunk 2]
```diff
@@ -154,3 +155,4 @@ sentinel failover-timeout <%=@name%> <%=@failovertimeout%>
 # Example:
 #
 # sentinel client-reconfig-script mymaster /var/redis/reconfig.sh
+<%= "sentinel client-reconfig-script #{@name} #{@client_reconfig_script}" unless @client_reconfig_script.nil? %>
```
Comment (martinb3) [inline templates/default/sentinel.conf.erb]: Should this just be guarded on `unless @client_reconfig_script.nil`?

**Comment by martinb3** [comment]
Do we need to guard this on a particular redis version? We sometimes have users set values expecting them to work, and then they are unhappy to discover we offered an option that won't work. For example, [this guard](https://github.com/brianbianco/redisio/blob/master/templates/default/redis.conf.erb#L445).

File: templates/default/sentinel.conf.erb (hunk 1 above)
Comment (matkam) [inline templates/default/sentinel.conf.erb]: Here's the commit notification-script support was added: https://github.com/antirez/redis/commit/baace5fc42dbe527a452d8864f37716d644f444a#diff-9cf0c48812ea7f4e98ef1281e344017f. Looks like Redis 2.6.

File: templates/default/sentinel.conf.erb (hunk 2 above)
Comment (matkam) [inline templates/default/sentinel.conf.erb]: client-reconfig-script added here: https://github.com/antirez/redis/commit/6275004ca6c5a127c84158a290e9524c5ef5c410#diff-9cf0c48812ea7f4e98ef1281e344017f
Looks like Redis 2.9.7

**Comment by matkam** [comment]
Do we have users on Sentinels pre Redis 2.8? Looks like support for client-reconfig-script and notification-script was added sometime between 2.6 and 2.8, 3 years ago.

**Comment by martinb3** [comment]
I think we do have folks using the cookbook with redis as old as 2.5. So I was just thinking we should probably do something like this in your template changes:
 https://github.com/brianbianco/redisio/blob/master/templates/default/redis.conf.erb#L445

Just a way to only add these configuration directives if the redis version is appropriate.

@zarry or @brianbianco -- I see we don't document what minimum redis versions we support. Perhaps we should do that, and then we can better evaluate PRs for compatibility? I'd nominate that we don't support older than the 2.6.x releases.

**Comment by matkam** [comment]
Added a guard to require Redis 2.8 and above @martinb3

**Comment by matkam** [comment]
Just another note, when merging this along with #193 will need to change ```name``` to ```mastername```

**Comment by miguelcnf** [comment]
:+1: for this feat.

@martinb3 any update?

**Comment by martinb3** [comment]
@miguelcnf I'm traveling until Monday, but I'll review and merge these then. We have a few others that also need to go in as well. Thanks!

**Comment by brianbianco** [comment]
This branch now has conflicts.  I haven't reviewed it but can we get the author to chime in or somebody who has reviewed it before like @martinb3 ?

**Comment by matkam** [comment]
@brianbianco just merged resolving conflicts, though I haven't had a chance to test yet.

**Comment by shortdudey123** [comment]
Carried by https://github.com/brianbianco/redisio/pull/342

**Pull request closed without merging**